src/charm.py

```diff
@@ -632,19 +714,10 @@ def _on_start(self, event) -> None:
         It is needed to install mongodb-clients inside the charm container
         to make this function work correctly.
         """
-        container = self.unit.get_container(Config.CONTAINER_NAME)
-        if not container.can_connect():
-            logger.debug("mongod container is not ready yet.")
-            event.defer()
-            return
-
-        if not container.exists(Config.SOCKET_PATH):
-            logger.debug("The mongod socket is not ready yet.")
-            event.defer()
-            return
+        if self.unit.is_leader() and not self.upgrade_in_progress:
+            self.upgrade._upgrade.set_versions_in_app_databag()
```

Comment:
Actually this might be a miss from my side, as it is run anyway as part of the `self.upgrade._reconcile_upgrade(event)` which is a few lines below.
I think I can safely remove this.